great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -298,12 +296,24 @@ def _init_factories(self) -> None:
 
     def _init_analytics(self) -> None:
         init_analytics(
+            enable=self._determine_analytics_enabled(),
             user_id=None,
-            data_context_id=uuid.UUID(self._data_context_id),
+            data_context_id=self._data_context_id,
             organization_id=None,
             oss_id=self._get_oss_id(),
         )
 
+    def _determine_analytics_enabled(self) -> bool:
+        # If the config file is missing the analytics_enabled key, we default to True
+        config_file_enabled = self.config.analytics_enabled
+        if config_file_enabled is None:
+            config_file_enabled = True
+
+        env_var_enabled = ENV_CONFIG.posthog_enabled
+
+        # A user must opt out through either the config file or env var to disable analytics
+        return config_file_enabled and env_var_enabled
```

Comment:
This logic seems wrong (and doesn't match the comment). `s/and/or`. I might move the comment up to a docstring too!